torch/csrc/distributed/autograd/context/dist_autograd_container.cpp

```diff
@@ -128,12 +134,27 @@ DistAutogradContext& DistAutogradContainer::currentContext() {
   return it->second;
 }
 
-void DistAutogradContainer::releaseContext(int64_t context_id) {
+void DistAutogradContainer::releaseContext(
+    int64_t context_id,
+    bool notifyWorkers) {
   std::lock_guard<std::mutex> guard(autograd_context_lock_);
+
   TORCH_CHECK(
       autograd_context_.find(context_id) != autograd_context_.end(),
       "Could not find autograd context with id: ",
       context_id);
+
+  if (notifyWorkers) {
+    // notify other workers to clean up their contexts.
+    auto workerIds =
+        autograd_context_.find(context_id)->second.getKnownWorkerIds();
+    auto agent = rpc::RpcAgent::getDefaultRpcAgent();
+    for (const auto& worker_id : workerIds) {
+      agent->send(
```

Comment:
Create a gh issue mentioning that we should store the futures we get from these calls in a structure and retry them later if they failed.